Makefile

```diff
@@ -169,6 +169,22 @@ build-ci: build-bacalhau install-plugins
 build-dev: build-ci
 	sudo cp ${BINARY_PATH} /usr/local/bin
 
+################################################################################
+# Target: build-webui
+################################################################################
+WEB_GO_FILES := $(shell find webui -name '*.go')
+WEB_SRC_FILES := $(shell find webui -not -path 'webui/build/*' -not -path 'webui/build' -not -path 'webui/node_modules/*' -not -name '*.go')
+WEB_BUILD_FILES := $(shell find webui/build) webui/build/index.html
+
+.PHONY: build-webui
+build-webui: ${WEB_BUILD_FILES}
+
+webui/build:
+	mkdir -p $@ && touch $@/.keep
+
+$(WEB_BUILD_FILES): $(WEB_SRC_FILES) webui/build
+	cd webui && npm run build
+
```

Comment:
There's no `npm install` step here, so on a fresh build this will fail to compile.

```
Makefile:186: warning: overriding commands for target `webui/build'
Makefile:183: warning: ignoring old commands for target `webui/build'
make: Circular webui/build <- webui/build dependency dropped.
cd webui && npm run build

> webui@0.1.0 build
> react-scripts build

sh: react-scripts: command not found
make: *** [webui/build] Error 127
```

Annoyingly `npm install` always seems to go to the internet, so it would be good to have if `node_modules` doesn't exist or the `package.json` has changes, then build.

Perhaps for now having a check for it the node_modules exists and if not, running `npm install`? That way we only have to do it once, and can delete the folder to rebuild if we add new modules.